File: 백준/Silver/15988. 1， 2， 3 더하기 3/1， 2， 3 더하기 3.py
T = int(input())
for tc in range(T):
    n = int(input())
    arr = [0] * (n+1)
    arr[1] = 1
    if n == 1:
        print(1)
        continue
    arr[2] = 2
    if n == 2:
        print(2)
        continue
    arr[3] = 4
    for i in range(4, n+1):
        arr[i] = (arr[i-1] + arr[i-2] + arr[i-3]) % 1000000009
    print(arr[n])

File: 백준/Silver/15988. 1， 2， 3 더하기 3/1， 2， 3 더하기 3.java
import java.util.Scanner;


public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();
        int[] arr = new int[T];

        for(int i = 0; i < T; i++) {
            arr[i] = sc.nextInt();
        }
        double[] ans = new double[1000001];
        ans[1] = 1;
        ans[2] = 2;
        ans[3] = 4;
        for(int j = 4; j < 1000000+1; j++){
            ans[j] = (ans[j-1]+ans[j-2]+ans[j-3]) % 1000000009;
        }
        for(int i = 0; i < T; i++){
            System.out.println((int) ans[arr[i]]);
        }

    }




   }